x-pack/plugins/alerting/server/task_runner/task_runner.test.ts

```diff
@@ -978,7 +979,128 @@ describe('Task Runner', () => {
+      ).actionsPlugin.isActionTypeEnabled.mockReturnValue(true);
+      customTaskRunnerFactoryInitializerParams.actionsPlugin.isActionExecutable.mockReturnValue(
+        true
+      );
+      actionsClient.ephemeralEnqueuedExecution.mockResolvedValue(mockRunNowResponse);
+      ruleType.executor.mockImplementation(
+        async ({
+          services: executorServices,
+        }: AlertExecutorOptions<
+          AlertTypeParams,
+          AlertTypeState,
+          AlertInstanceState,
+          AlertInstanceContext,
+          string
+        >) => {
+          executorServices.alertInstanceFactory('1').scheduleActions('default');
+          executorServices.alertInstanceFactory('2').scheduleActions('default');
+        }
+      );
+      const taskRunner = new TaskRunner(
+        ruleType,
+        {
+          ...mockedTaskInstance,
+          state: {
+            ...mockedTaskInstance.state,
+            alertInstances: {
+              '2': {
+                meta: {
+                  lastScheduledActions: { date: moment().toISOString(), group: 'default' },
+                },
+                state: {
+                  bar: false,
+                  start: '1969-12-31T00:00:00.000Z',
+                  duration: 86400000000000,
+                },
+              },
+            },
+          },
+        },
+        taskRunnerFactoryInitializerParams
+      );
+      rulesClient.get.mockResolvedValue({
+        ...mockedRuleTypeSavedObject,
+        throttle: '1d',
+      });
+      encryptedSavedObjectsClient.getDecryptedAsInternalUser.mockResolvedValue({
+        id: '1',
+        type: 'alert',
+        attributes: {
+          apiKey: Buffer.from('123:abc').toString('base64'),
+          enabled: true,
+        },
+        references: [],
+      });
+      await taskRunner.run();
+      // expect(enqueueFunction).toHaveBeenCalledTimes(1);
+
+      const logger = customTaskRunnerFactoryInitializerParams.logger;
+      // expect(logger.debug).toHaveBeenCalledTimes(5);
+      expect(logger.debug).nthCalledWith(
+        3,
+        `skipping scheduling of actions for '2' in rule test:1: 'rule-name': rule is throttled`
+      );
+    }
+  );
+
+  test.each(ephemeralTestParams)(
+    'skips firing actions for active alert when alert is muted even if notifyWhen === onActionGroupChange %s',
+    async (nameExtension, customTaskRunnerFactoryInitializerParams, enqueueFunction) => {
+      customTaskRunnerFactoryInitializerParams.actionsPlugin.isActionExecutable.mockReturnValue(
+        true
+      );
+      ruleType.executor.mockImplementation(
+        async ({
+          services: executorServices,
+        }: AlertExecutorOptions<
+          AlertTypeParams,
+          AlertTypeState,
+          AlertInstanceState,
+          AlertInstanceContext,
+          string
+        >) => {
+          executorServices.alertInstanceFactory('1').scheduleActions('default');
```

Comment:
Same comment as above

Reply:
yes there is already a warning about the merge conflict, i have to update this :)